e2e/browser/src/pages/index.jsx

```diff
@@ -20,14 +20,125 @@
  */
 
 import { useSession } from "@inrupt/solid-ui-react";
+import { useEffect, useState } from "react";
+import {
+  approveAccessRequest,
+  revokeAccessGrant,
+} from "@inrupt/solid-client-access-grants";
+import {
+  getPodUrlAll,
+  saveFileInContainer,
+  getSourceUrl,
+  deleteFile,
+} from "@inrupt/solid-client";
+
+// This is the content of the file uploaded manually at SHARED_FILE_IRI.
+const SHARED_FILE_CONTENT = "Some content.\n";
 
 export default function Home() {
   const { session } = useSession();
+  const [accessGrant, setAccessGrant] = useState(undefined);
+  const [sharedResourceIri, setSharedResourceIri] = useState(undefined);
+
+  const handleCreate = (e) => {
+    // This prevents the default behaviour of the button, i.e. to resubmit, which reloads the page.
+    e.preventDefault();
+    if (typeof sharedResourceIri === "string") {
+      // If a resource already exist, do nothing
+      return;
+    }
+    (async () => {
+      // Create a file in the resource owner's Pod
+      const resourceOwnerPodAll = await getPodUrlAll(session.info.webId);
+      if (resourceOwnerPodAll.length === 0) {
+        throw new Error(
+          "The Resource Owner WebID Profile is missing a link to at least one Pod root."
+        );
+      }
+      const savedFile = await saveFileInContainer(
+        resourceOwnerPodAll[0],
+        Buffer.from(SHARED_FILE_CONTENT),
+        {
+          // The session ID is a random string, used here as a unique slug.
+          slug: `${session.info.sessionId}.txt`,
+          fetch: session.fetch,
+        }
+      );
+      setSharedResourceIri(getSourceUrl(savedFile));
+    })();
+  };
+
+  const handleDelete = (e) => {
+    // This prevents the default behaviour of the button, i.e. to resubmit, which reloads the page.
+    e.preventDefault();
+    (async () => {
+      await deleteFile(sharedResourceIri, {
+        fetch: session.fetch,
+      });
+      setSharedResourceIri(undefined);
+    })();
+  };
+
+  const handleGrant = (e) => {
+    // This prevents the default behaviour of the button, i.e. to resubmit, which reloads the page.
+    e.preventDefault();
+    if (typeof sharedResourceIri !== "string") {
+      // If the resource does not exist, do nothing.
+      return;
+    }
+    (async () => {
+      const accessGrant = await approveAccessRequest(
+        session.info.webId,
+        undefined,
+        {
+          requestor: "https://johndoe.webid",
```

Comment:
Don't we need a "real" webid here?

As in, is there no WebID verification step?

Reply:
No, there is no verification step when granting access, the verification would happen when trying to exercise this grant. Similarly, you could add any arbitrary IRI in an ACP matcher, there is no verification step at this point in the process.